app/src/main/kotlin/com/simplemobiletools/calendar/pro/activities/EventActivity.kt

```diff
@@ -240,11 +240,27 @@ class EventActivity : SimpleActivity() {
             R.id.delete -> deleteEvent()
             R.id.duplicate -> duplicateEvent()
             R.id.share -> shareEvent()
+            android.R.id.home -> onBackPressed()
             else -> return super.onOptionsItemSelected(item)
         }
         return true
     }
 
+    override fun onBackPressed() {
+        CloseEditingDialog(this) {
+            ensureBackgroundThread {
+                when (it) {
+                    CLOSE_WITHOUT_SAVING -> {
+                        runOnUiThread {
```

Comment:
why are you creating a new thread above, then switching back to UI thread without doing anything? Create a new thread only at saving an event